kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/records/BaseConsumerRecord.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.clients.records;
+
+import java.util.Optional;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Headers;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.systemtests.utils.KafkaUtils;
+
+public class BaseConsumerRecord {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseConsumerRecord.class);
+
+    protected String topic;
+    protected long timestamp;
+    protected String timestampType;
+    protected Object key;
+    protected Object payload;
+    protected int partition;
+    protected long offset;
+    protected Integer leaderEpoch;
+    protected Headers recordHeaders;
+
+    /**
+     * Parse from json string t.
+     *
+     * @param <T>  the type parameter
+     * @param valueTypeRef the value type ref
+     * @param response the response
+     * @return the t
+     */
+    public static <T> T parseFromJsonString(TypeReference<T> valueTypeRef, String response) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            return objectMapper.readValue(response, valueTypeRef);
+        }
+        catch (JsonProcessingException e) {
+            LOGGER.atError().setMessage("Something bad happened").setCause(e).log();
```

Comment:
We should think early if deserialising goes wrong, to just rethrow an UncheckedIOException wrapping the JPE.

Reply:
The reason I don't throw an exception is because sometimes there are some extra unexpected logs with no JSON format, and it would cause a failure when it is not.

this is an example:
```
%reconnected to offset 0: exiting
{"topic":"my-topic-c3ca47","partition":0,"offset":0,"tstype":"create","ts":1717412015716,"broker":2,"key":null,"payload":"Hello-world - 0"}
{"topic":"my-topic-c3ca47","partition":0,"offset":1,"tstype":"create","ts":1717412015992,"broker":2,"key":null,"payload":"Hello-world - 1"}
```